### PR TITLE
implement malloc volume

### PR DESCRIPTION
**Pull request by moshe-shahar**

### CODE REVIEW

File: pkg/backend/malloc.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (c) 2022-2024 Dell Inc, or its subsidiaries.
+// Copyright (C) 2023 Intel Corporation
```
Comment (artek-koltun) [inline pkg/backend/malloc.go:3]: copyrights?

File: pkg/backend/malloc.go [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+		log.Printf("Already existing MallocVolume with id %v", in.MallocVolume.Name)
+		return volume, nil
+	}
+	// not found, so create a new one
+	params := spdk.BdevMallocCreateParams{
+		Name:         resourceID,
+		BlockSize:    int(in.GetMallocVolume().GetBlockSize()),
+		NumBlocks:    int(in.GetMallocVolume().GetBlocksCount()),
+		MdSize:       int(in.GetMallocVolume().GetMetadataSize()),
+		MdInterleave: true,
```
Comment (artek-koltun) [inline pkg/backend/malloc.go:59]: by default in SPDK it is false. Why do we need to have tn as `true` here?

File: pkg/backend/malloc_test.go [hunk 3]
```diff
@@ -0,0 +1,844 @@
...
+			out:     nil,
+			spdk:    []string{`{"id":%d,"error":{"code":1,"message":"myopierr"},"result":false}`},
+			errCode: codes.Unknown,
+			errMsg:  fmt.Sprintf("bdev_malloc_delete: %v", "json response error: myopierr"),
+			missing: false,
+		},
+		"valid request with valid SPDK response": {
+			in:      testMallocVolumeName,
+			out:     &emptypb.Empty{},
+			spdk:    []string{`{"id":%d,"error":{"code":0,"message":""},"result":true}`}, // `{"jsonrpc": "2.0", "id": 1, "result": True}`,
```
Comment (artek-koltun) [inline pkg/backend/malloc_test.go:778]: is there comment needed

**Review by artek-koltun — COMMENTED** [review]
Could you pls also update https://github.com/opiproject/opi-spdk-bridge/blob/main/scripts/tests.sh ? it uses some precreated mallocs, but maybe we could use our API to create our own mallocs to work with?

File: pkg/backend/malloc.go (hunk 2 above)
Comment (moshe-shahar) [reply pkg/backend/malloc.go:59]: I need it to be `interleaved`.
If you think it's wrong in that way, I will add it as another field.

**Comment by moshe-shahar** [comment]
> Could you pls also update https://github.com/opiproject/opi-spdk-bridge/blob/main/scripts/tests.sh ? it uses some precreated mallocs, but maybe we could use our API to create our own mallocs to work with?

I updated that. but I'm not sure it's working, so you can wait with the review.

**Comment by moshe-shahar** [comment]
CI passed. please review.

File: pkg/backend/malloc.go (hunk 2 above)
Comment (glimchb) [reply pkg/backend/malloc.go:60]: I need to read what interleave means…

**Review by glimchb — APPROVED** [review]
Lgtm

File: pkg/backend/malloc.go (hunk 2 above)
Comment (moshe-shahar) [reply pkg/backend/malloc.go:60]: Where the bdev keeps the MD. After each block (interleave) or on a dedicated region.

**Pull request merged**